### PR TITLE
chore!: Add last PSS changes

### PR DESCRIPTION
**Pull request by sainoe**
Please go to the `Preview` tab and select the appropriate sub-template:

* [Production code](?expand=1&template=production.md) - for types `fix`, `feat`, and `refactor`.
* [Docs](?expand=1&template=docs.md) - for documentation changes.
* [Others](?expand=1&template=others.md) - for changes that do not affect production code.

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected computation of the maximum capped power in the system.

- **Dependencies**
  - Updated `ibc-go` to version `v7.5.1`.

- **Documentation**
  - Added ADR 017 to allow inactive validators on the provider chain to validate on consumer chains.
  - Introduced a mechanism to cap the size of the validator set on a consumer chain.

- **Chores**
  - Updated Docker login action to version `v3.2.0`.
  - Updated `buf-setup-action` to version `v1.32.2`.

- **New Features**
  - Added `ConsumerModificationProposal` message for governance proposals to modify consumer chain parameters.
  - Introduced new RPC method `ConsumerModification` for modifying running consumer chains.

- **Tests**
  - Added test scenarios for handling consumer modification proposals.

- **Refactor**
  - Improved error handling and logging in various functions related to validator management and token allocation.

### CODE REVIEW

File: x/ccv/provider/keeper/keeper_test.go [hunk 1]
```diff
@@ -411,10 +412,12 @@ func TestGetAllConsumerChains(t *testing.T) {
...
 	}
 	powers := []int64{50, 150, 300, 500} // sum = 1000
-	mocks.MockStakingKeeper.EXPECT().GetLastValidators(gomock.Any()).Return(vals).AnyTimes()
+	mocks.MockStakingKeeper.EXPECT().GetLastValidators(gomock.Any()).Return(vals, nil).AnyTimes()
 
 	for i, val := range vals {
-		mocks.MockStakingKeeper.EXPECT().GetLastValidatorPower(gomock.Any(), val.GetOperator()).Return(powers[i]).AnyTimes()
+		valAddr, err := sdk.ValAddressFromBech32(val.GetOperator())
+		require.NoError(t, err)
+		mocks.MockStakingKeeper.EXPECT().GetLastValidatorPower(gomock.Any(), valAddr).Return(powers[i], nil).AnyTimes()
```
[the remark this answers is not in the document]
Comment (sainoe) [reply x/ccv/provider/keeper/keeper_test.go:420]: See line 403. The controller finish function is called within a defer statement.

File: x/ccv/provider/keeper/msg_server.go [hunk 2]
```diff
@@ -308,3 +303,17 @@
...
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := k.Keeper.HandleConsumerModificationProposal(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrapf(err, "failed handling ConsumerAddition proposal")
+	}
+
+	return &types.MsgConsumerModificationResponse{}, nil
+}
```
[the remark this answers is not in the document]
Comment (sainoe) [reply x/ccv/provider/keeper/msg_server.go:319]: Nice catch!

**Review by p-offtermatt — COMMENTED** [review]
LGTM in principle, could you look into the gosec failure?

**Comment by sainoe** [comment]
Gosec fails due to some low severity issues. All are unhandled errors and most of them are located in the genesis export. Not, concerning.

**Review by p-offtermatt — APPROVED** [review]
Thx for the work!

**Pull request merged**